### PR TITLE
Integrating the tokens packages into the tokens api

### PR DESCRIPTION
**Pull request by romellogoodman**
### WHY are these changes introduced?

Fixes #0000 

Removing the process of querying token values via github api and instead reading them from the package.

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
- [ ] For visual design changes, ping @ sarahill to update the Polaris UI kit

### CODE REVIEW

File: polaris.shopify.com/package.json [hunk 1]
```diff
@@ -10,8 +10,8 @@
   },
   "dependencies": {
     "@shopify/polaris": "^9.0.0",
+    "@shopify/polaris-tokens": "5.0.0-rc.1",
```
Comment (aaronccasanova) [inline polaris.shopify.com/package.json:13]: Friendly reminder to release this RC before merging this PR. (I don't see it on [unpkg](https://unpkg.com/browse/@shopify/polaris-tokens@5.0.0-rc.1))

File: polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx [hunk 2]
```diff
@@ -67,28 +34,22 @@ function isFormat(format: unknown): format is Format {
...
 /**
  * Format the token data into: css or json
  */
 const formatTokenGroup = (tokenGroup: TokenGroup, format: Format) => {
+  const tokenValues = Object.fromEntries(
+    Object.entries(tokenGroup).map((entry) => {
+      const [key, tokenProps] = entry as [TokenGroupKey, TokenProperties];
+
+      return [key, tokenProps.value];
+    }),
```
Comment (aaronccasanova) [inline polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx]:
```suggestion
    Object.entries(tokenGroup).map(([token, tokenProps]) => [token, tokenProps.value]),
```

After jamming this morning I realized we can probably clean this up a bit.

File: polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx [hunk 3]
```diff
@@ -67,28 +34,22 @@ function isFormat(format: unknown): format is Format {
...
+  );
+
   if (format === 'css') {
-    return Object.keys(tokenGroup)
+    return Object.keys(tokenValues)
       .reduce<string[]>((result, token) => {
-        const cssVariable = `--${token}: ${tokenGroup[token]};`;
+        const cssVariable = `--${token}: ${tokenValues[token]};`;
```
Comment (aaronccasanova) [inline polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx]:
```suggestion
        const cssVariable = `--p-${token}: ${tokenValues[token]};`;
```

Should this have a `p-`? I think it's time we create a `getVar` or `createVar` util that accepts a `tokenName` and generates the `--p-<tokenName>` string. Happy for that to be in a separate PR, but we are doing that transformation in multiple places and would be nice to have an established pattern.

File: polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx [hunk 4]
```diff
@@ -97,67 +58,47 @@ const formatTokenGroup = (tokenGroup: TokenGroup, format: Format) => {
       .join('\n');
   }
 
-  return tokenGroup;
+  return tokenValues;
 };
 
 const handler = async (req: NextApiRequest, res: NextApiResponse) => {
   const formatParam = (req.query.format || 'json') as Format;
```
Comment (aaronccasanova) [inline polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx]:
```suggestion
  const formatParam = isFormat(req.query.format) ? req.query.format : 'json';
```

We already have a `isFormat` type guard. Perhaps, we should use that here so users can't define an arbitrary format which TypeScript is currently casting to the `Format` type. Thoughts?

File: polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx [hunk 5]
```diff
@@ -97,67 +58,47 @@ const formatTokenGroup = (tokenGroup: TokenGroup, format: Format) => {
...
-  const schemeParam = (req.query.scheme || 'light') as Scheme;
+  const schemeParam = (req.query.scheme || 'light') as ColorScheme;
```
Comment (aaronccasanova) [inline polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx]:
```suggestion
  const schemeParam = isScheme(req.query.scheme) ? req.query.scheme : 'light';
```

Same as above. This PR removes the `isScheme` type guard which safely checks and casts the `query.scheme` param. Thoughts?

File: polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx [hunk 6]
```diff
@@ -97,67 +58,47 @@ const formatTokenGroup = (tokenGroup: TokenGroup, format: Format) => {
...
 
   if (typeof formatParam === 'string' && typeof schemeParam === 'string') {
-    const colorSchemeTokenGroup = colorSchemeMap[schemeParam];
-    const tokenGroupParam: any = req.query.tokens || '';
-    let tokenGroups: string[] = [];
+    const colorScheme = osColorSchemes[schemeParam] as ColorScheme;
```
Comment (aaronccasanova) [inline polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx]:
```suggestion
```
`osColorSchemes` are used for setting the associated [`color-scheme` CSS property](https://github.com/Shopify/polaris/blob/5741343151f3ec52842878d8ce607cd8d9fc0053/polaris-react/src/components/CustomProperties/styles.ts#L64). You shouldn't need that value anywhere in this file and should instead use the `schemeParam` (if I understand correctly).

File: polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx [hunk 7]
```diff
@@ -97,67 +58,47 @@ const formatTokenGroup = (tokenGroup: TokenGroup, format: Format) => {
...
+        tokenData = {...tokenData, ...tokenGroup};
+      });
     }
 
-    if (staticTokenGroupKeys.includes(tokenGroupParam)) {
-      tokenGroups.push(tokenGroupParam);
-    }
+    if (tokenGroupParam === 'all' || tokenGroupParam === 'colors') {
+      const colorSchemeTokenGroup: TokenGroup =
+        tokens.colorSchemes[colorScheme] || {};
```
Comment (aaronccasanova) [inline polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx]:
```suggestion
        tokens.colorSchemes[schemeParam] || {};
```
As mentioned above, we should use the `schemeParam` to access the `colorSchemes` token group.

File: polaris.shopify.com/pages/api/v0/tokens/index.tsx [hunk 8]
```diff
@@ -62,16 +66,14 @@ const html = `
                 </tr>
             </thead>
             <tbody>
-            ${Object.keys(colorSchemeMap)
-              .map((colorScheme) => {
-                const scheme = colorScheme as Scheme;
+            ${Object.keys(osColorSchemes)
+              .map((scheme) => {
```
Comment (aaronccasanova) [inline polaris.shopify.com/pages/api/v0/tokens/index.tsx]: Same as above, https://github.com/Shopify/polaris/pull/5537#discussion_r852301564. I suggestion either exporting the [`colorSchemes` token groups](https://github.com/Shopify/polaris/blob/5741343151f3ec52842878d8ce607cd8d9fc0053/polaris-tokens/tokens.ts#L49-L52) from `polaris-tokens` or accessing them from `tokens` e.g.
```js
import {tokens} from '@shopify/polaris-tokens'
Object.keys(tokens.colorSchemes).map(...)
```

File: polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx (hunk 2 above)
Comment (romellogoodman) [reply polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx]: Nice added locally and pushed up!

File: polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx (hunk 3 above)
Comment (romellogoodman) [reply polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx]: Created https://github.com/Shopify/polaris/pull/5540

File: polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx [hunk 9]
```diff
@@ -97,67 +60,46 @@ const formatTokenGroup = (tokenGroup: TokenGroup, format: Format) => {
...
-  const formatParam = (req.query.format || 'json') as Format;
-  const schemeParam = (req.query.scheme || 'light') as Scheme;
+  const formatParam = isFormat(req.query.format) ? req.query.format : 'json';
+  const schemeParam = isScheme(req.query.scheme) ? req.query.scheme : 'light';
 
   if (typeof formatParam === 'string' && typeof schemeParam === 'string') {
-    const colorSchemeTokenGroup = colorSchemeMap[schemeParam];
-    const tokenGroupParam: any = req.query.tokens || '';
-    let tokenGroups: string[] = [];
+    const tokenGroupParam = (req.query.tokens || '') as TokenGroupKey;
```
Comment (aaronccasanova) [inline polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx]:
```suggestion
    const tokenGroupParam = req.query.tokens || '';
```

I think we can remove this cast now with the updated type guards.

File: polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx [hunk 10]
```diff
@@ -97,67 +60,46 @@ const formatTokenGroup = (tokenGroup: TokenGroup, format: Format) => {
...
-    try {
-      requests = tokenGroups.map((tokenFile) => {
-        const url = getGithubUrl(tokenFile, true);
-
-        if (url === null) {
-          throw new Error('Could not fetch tokens');
-        }
+    if (staticTokenGroupKeys.includes(tokenGroupParam as StaticTokenGroupKey)) {
+      const tokenGroupName = tokenGroupParam as StaticTokenGroupKey;
+      const tokenGroup: TokenGroup = tokens[tokenGroupName] || {};
```
Comment (aaronccasanova) [inline polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx]:
```suggestion
    if (isStaticTokenGroupKey(tokenGroupParam)) {
      const tokenGroup: TokenGroup = tokens[tokenGroupParam];
```

**suggestion (non-blocking):** If we keep the above `isStaticTokenGroupKey` type guard we can avoid the cast and remove the fallback.

File: polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx [hunk 11]
```diff
@@ -1,64 +1,30 @@
...
   'depth',
-  'legacy-tokens',
+  'legacyTokens',
   'motion',
   'shape',
   'spacing',
   'typography',
-  'z-index',
+  'zIndex',
 ] as const;
```
Comment (aaronccasanova) [inline polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx]: Perhaps we should derive this data structure now that we have direct access to the tokens e.g.
```ts
import {tokens, Tokens} from '@shopify/polaris-tokens';

type StaticTokenGroupKey = Exclude<keyof Tokens, 'colorSchemes'>;

const staticTokenGroupKeys = Object.keys(tokens).filter(
  (token) => token !== 'colorSchemes',
) as StaticTokenGroupKey[];
```

File: polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx [hunk 12]
```diff
@@ -97,67 +60,46 @@ const formatTokenGroup = (tokenGroup: TokenGroup, format: Format) => {
...
-    let tokenGroups: string[] = [];
+    const tokenGroupParam = (req.query.tokens || '') as TokenGroupKey;
+    let tokenData: TokenGroup = {};
 
     // Determine which list(s) we are querying for based on the token param
     if (tokenGroupParam === 'all') {
-      tokenGroups = tokenGroups.concat(staticTokenGroupKeys);
-    }
+      staticTokenGroupKeys.forEach((group) => {
+        const tokenGroup: TokenGroup = tokens[group] || {};
```
Comment (aaronccasanova) [inline polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx]:
```suggestion
        const tokenGroup = tokens[group];
```
Do we need the fallback or the explicit type?

File: polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx [hunk 13]
```diff
@@ -97,67 +60,46 @@ const formatTokenGroup = (tokenGroup: TokenGroup, format: Format) => {
...
+        tokenData = {...tokenData, ...tokenGroup};
+      });
     }
 
-    if (staticTokenGroupKeys.includes(tokenGroupParam)) {
-      tokenGroups.push(tokenGroupParam);
-    }
+    if (tokenGroupParam === 'all' || tokenGroupParam === 'colors') {
+      const colorSchemeTokenGroup: TokenGroup =
+        tokens.colorSchemes[schemeParam] || {};
```
Comment (aaronccasanova) [inline polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx]:
```suggestion
      const colorSchemeTokenGroup = tokens.colorSchemes[schemeParam];
```

Same as above. The conditionals should guarantee this member access and infer the `TokenGroup` type.

**Review by aaronccasanova — APPROVED** [review]
Updates look great! Left a few more clean up suggestions (all non-blocking). Approving with the assumption the `@shopify/polaris-tokens` RC will be released before merging.

**Pull request merged**